AppCenterData/AppCenterData/MSData.m

```diff
@@ -638,6 +652,23 @@ - (MSDataError *)generateInvalidClassError {
   return dataError;
 }
 
+- (BOOL)isDocumentIdInvalid:(NSString *)documentId {
+  if (!documentId) {
+    return YES;
+  }
+  NSRegularExpression *expr = [NSRegularExpression regularExpressionWithPattern:kMSDocumentIdValidationPattern options:0 error:nil];
+  NSUInteger matchCount = [expr numberOfMatchesInString:documentId options:0 range:NSMakeRange(0, documentId.length)];
+  return matchCount == 0;
+}
+
+- (MSDataError *)generateInvalidDocumentIdError {
+  MSDataError *dataError = [[MSDataError alloc] initWithErrorCode:MSACDataErrorDocumentIdInvalid
+                                                       innerError:nil
+                                                          message:(NSString *)kMSACDataErrorDocumentIdInvalidDesc];
+  MSLogError([MSData logTag], @"Document ID is invalid.");
+  return dataError;
```

Comment:
Is it ok that declared return type does not match actually returned value type? I'd prefer explicit conversion over implicit

Reply:
Not sure what you mean since they are both `MSDataError`